src/storage/metrics.rs

```diff
@@ -137,3 +137,29 @@ lazy_static! {
             exponential_buckets(1.0, 2.0, 21).unwrap()
         ).unwrap();
 }
+
+pub struct LocalCounter {
+    counter: Counter,
+    val: f64,
+}
+
+impl LocalCounter {
+    pub fn new(counter: Counter) -> LocalCounter {
+        LocalCounter {
+            counter: counter,
+            val: 0.0,
+        }
+    }
+
+    /// `inc` increments the local counter by 1.
+    #[inline]
+    pub fn inc(&mut self) {
+        self.val += 1.0;
+    }
+
+    /// `flush` the local counter value to the counter
+    #[inline]
+    pub fn flush(&self) -> Result<()> {
+        self.counter.inc_by(self.val)
```

Comment:
After flush, we should clean up the value. 
No need to return Result, using unwrap here is fine.

Reply:
if val is 0, no need to flush to global counter.